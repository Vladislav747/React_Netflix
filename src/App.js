import React from "react";
import "./App.css";
import Row from "./components/Row";
import Banner from "./components/Banner/Banner";
import request from "./requests";

function App() {
  return (
    <div className="App">
      <h1>Hey Clever Programmer! Let's build</h1>
      <Banner />
      <Row
        title="NETFLIX ORIGINALS"
        fetchUrl={request.fetchNetflixOriginals}
        isLargeRow={true}
      />
      <Row title="Trending Now" fetchUrl={request.fetchTrending} />
      <Row title="Top Rated" fetchUrl={request.fetchTopRated} />
      <Row title="Action Movies" fetchUrl={request.fetchActionMovies} />
      <Row title="Comedy Movies" fetchUrl={request.fetchComedyMovies} />
      <Row title="Horror Movies" fetchUrl={request.fetchHorrorMovies} />
      <Row title="Romance Movies" fetchUrl={request.fetchRomanceMovies} />
      <Row title="Documentaries" fetchUrl={request.fetchDocumentaries} />
    </div>
  );
}

export default App;
