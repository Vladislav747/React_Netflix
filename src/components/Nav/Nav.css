.nav {
  position: fixed;
  top: 0;
  display: flex;
  justify-content: space-between;
  z-index: 999;
}

.nav__logo {
  object-fit: contain;
  position: fixed;
  left: 20px;
  width: 80px;
}

.nav__avatar {
  position: fixed;
  object-fit: contain;
  width: 30px;
  right: 20px;
  padding: 20px;
}
