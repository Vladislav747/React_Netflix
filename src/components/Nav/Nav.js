import React, { useState, useEffect } from "react";

import "./Nav.css";

function Nav() {
  return (
    <div className="nav">
      <img
        className="nav__logo"
        src="http://static.wixstatic.com/media/cbafe1_63c8e817ce56404193de0db66bc66418~mv2_d_1600_1600_s_2.png"
        alt="Netflix Logo"
      />
      {/* <img
        className="nav__avatar"
        src="http://static.wixstatic.com/media/cbafe1_63c8e817ce56404193de0db66bc66418~mv2_d_1600_1600_s_2.png"
        alt="Netflix Avatar"
      /> */}
    </div>
  );
}

export default Nav;
