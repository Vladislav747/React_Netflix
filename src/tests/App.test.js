import React from "react";
import ReactDOM from "react-dom";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

import App from "../App";
import Nav from "../components/Nav/Nav";

configure({ adapter: new Adapter() });

describe("Render App", () => {
  it("render correctly App component", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
  });

  it("match snapshot", () => {
    let appComponent = renderer.create(<App />).toJSON();
    expect(appComponent).toMatchSnapshot();
  });
});
