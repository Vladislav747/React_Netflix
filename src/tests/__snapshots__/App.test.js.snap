// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App match snapshot 1`] = `
<div
  className="app"
>
  <div
    className="nav false"
  >
    <img
      alt="Netflix Logo"
      className="nav__logo"
      src="http://static.wixstatic.com/media/cbafe1_63c8e817ce56404193de0db66bc66418~mv2_d_1600_1600_s_2.png"
    />
  </div>
  <header
    className="banner"
    style={
      Object {
        "backgroundImage": "url(\\"https://image.tmdb.org/t/p/original/undefined\\")",
      }
    }
  >
    <div
      className="banner__contents"
    >
      <h1 />
      <div
        className="banner__buttons"
      >
        <button
          className="banner__button"
        >
          Play
        </button>
        <button
          className="banner__button"
        >
          Play
        </button>
      </div>
      <h1
        className="banner__description"
      />
    </div>
    <div
      className="banner--fadeBottom"
    />
  </header>
  <div
    className="row"
  >
    <h2>
      NETFLIX ORIGINALS
    </h2>
    <div
      className="row__posters"
    />
  </div>
  <div
    className="row"
  >
    <h2>
      Trending Now
    </h2>
    <div
      className="row__posters"
    />
  </div>
  <div
    className="row"
  >
    <h2>
      Top Rated
    </h2>
    <div
      className="row__posters"
    />
  </div>
  <div
    className="row"
  >
    <h2>
      Action Movies
    </h2>
    <div
      className="row__posters"
    />
  </div>
  <div
    className="row"
  >
    <h2>
      Comedy Movies
    </h2>
    <div
      className="row__posters"
    />
  </div>
  <div
    className="row"
  >
    <h2>
      Horror Movies
    </h2>
    <div
      className="row__posters"
    />
  </div>
  <div
    className="row"
  >
    <h2>
      Romance Movies
    </h2>
    <div
      className="row__posters"
    />
  </div>
  <div
    className="row"
  >
    <h2>
      Documentaries
    </h2>
    <div
      className="row__posters"
    />
  </div>
</div>
`;
